components/kyma-environment-broker/internal/orchestration/manager/manager.go

```diff
@@ -199,9 +188,16 @@ func (m *orchestrationManager) NewOperationForPendingRetrying(o *internal.Orches
 			return nil, o, len(runtimes), fmt.Errorf("while getting instance %s: %w", r.InstanceID, err)
 		}
 
-		op, err := m.factory.NewOperation(*o, r, *inst, orchestration.Pending)
+		exist, op, err := m.factory.QueryOperation(o.OrchestrationID, r)
 		if err != nil {
-			return nil, o, len(runtimes), fmt.Errorf("while creating new operation for runtime id %q: %w", r.RuntimeID, err)
+			return nil, o, len(runtimes), fmt.Errorf("while quering operation for runtime id %q: %w", r.RuntimeID, err)
+		}
+
+		if !exist {
+			op, err = m.factory.NewOperation(*o, r, *inst, orchestration.Pending)
+			if err != nil {
+				return nil, o, len(runtimes), fmt.Errorf("while creating new operation for runtime id %q: %w", r.RuntimeID, err)
+			}
```

Comment:
This querying and checking for existence should be done when orchestration is in `Pending` state (i.e. it's not `Retrying`).
Also, it should be the first thing when looping through the runtime operations, and the above if/else code should also not be executed (skipped when exists, no need to resolve maintenance window time and get instance).

Reply:
fixed as suggested